pyrubberband/pyrb.py

```diff
@@ -141,6 +141,66 @@ def time_stretch(y, sr, rate, rbargs=None):
     return __rubberband(y, sr, **rbargs)
 
 
+def timemap_stretch(y, sr, time_map, rbargs=None):
+    '''Apply a timemap stretch to an audio time series.
+
+    This uses the `time` and `timemap` form for rubberband.
+
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,) or (n, c)]
+        Audio time series, either single or multichannel
+
+    sr : int > 0
+        Sampling rate of `y`
+
+    time_map : list
+        Each element is a tuple `t` of length 2 which correspond to the input
+         frame and desired output frame.
+        If t[1] < t[0] the track will be sped up in this area.
+        time_map[-1] must correspond to the lengths of the input audio and
+         output audio.
+
+    rbargs
+        Additional keyword parameters for rubberband
+
+        See `rubberband -h` for details.
+
+    Returns
+    -------
+    y_stretch : np.ndarray
+        Time-stretched audio
+
+    Raises
+    ------
+    ValueError
+        if `rate <= 0`
+    '''
+
+    if rbargs is None:
+        rbargs = dict()
+
+    time_stretch = time_map[-1][1] * 1.0 / time_map[-1][0]
+
+    rbargs.setdefault('--time', time_stretch)
+
+    fs, stretch_file_name = tempfile.mkstemp(suffix='.txt')
+    os.close(fs)
```

Comment:
Why close this file, reopen it (with a new descriptor), write, close, and then unlink?  It would be cleaner to use a `tempfile.NamedTemporaryFile` context manager to handle all the cleanup in case of errors.

Reply:
It feels like the easiest solution is:
```
stretch_file = tempfile.NamedTemporaryFile(mode='w', suffix='.txt', delete=False)
for t in time_map:
    stretch_file.write('{:0} {:1}\n'.format(t[0], t[1])) 
stretch_file.close()
# run the rubberband command
os.unlink(stretch_file.name)
```

I tried the following but it does not work
```
stretch_file = tempfile.NamedTemporaryFile(mode='w', suffix='.txt')
for t in time_map:
    stretch_file.write('{:0} {:1}\n'.format(t[0], t[1])) 
# run the rubberband command
stretch_file.close()
```

Reply:
Hmm.  In that case, it should be wrapped in a try:... finally: so that the unlink cleanup always happens.